maps/RandomZLevels/tomb.dm

```diff
@@ -260,3 +263,63 @@
 		AM.ex_act(1)
 
 	explosion(T, -1, -1, 1)
+
+
+//SPECIAL BUTTONS
+//Only two can be activated at once
+//Activating a third button when two are activated toggles the first one off
+/obj/structure/button/water_puzzle
+	var/global/list/last_pressed = list()
+
+	global_search = 0 //Only current area
+
+/obj/structure/button/water_puzzle/Destroy()
+	last_pressed.Remove(src)
+	..()
+
+/obj/structure/button/water_puzzle/activate(force = 0)
+	if(state == 1) //Activated
+		if(!force)
+			return
+
+		return ..()
+
+	else if(last_pressed.len == 2) //Not activated - turning the button on
+		var/obj/structure/button/water_puzzle/button_to_toggle_off = last_pressed[1]
+		if(button_to_toggle_off.state == 1)
+			button_to_toggle_off.activate(1) //Force the first activated button to turn off
+			last_pressed.Remove(button_to_toggle_off)
+
+	..()
+	last_pressed.Add(src)
+
+/obj/structure/button/water_puzzle/is_valid_door(obj/effect/hidden_door/D)
+	return (..() || (D.icon_state == "wildcard")) //Activate wildcard doors too
+
+/obj/item/weapon/paper/tomb_notes
+	name = "paper- 'My Notes'"
+	info = {"<i>You can't go through this room without a partner, so I can't advance any further. I hope these notes will help you.<BR>
+	The water is powered by magic, there is no better explanation for its behaviour. I can't touch it or jump into it. In the water there are metal platforms that you can walk on. There are also 6 buttons on the wall.<BR>
+	There are also 7 groups of platforms, one for each button, plus to one rogue group. Pressing a button raises its group of platforms above the water. Only two platform groups can be raised at once; pressing a third button will cause one group to lower. I think the one which was raised the earlier is lowered, but maybe not.<BR>
+	There are <s>5 6</s> 7 rogue platforms, they are lowered and raised whenever a button is pressed. Any button. You may want to find them immediately, because they look exactly like normal platforms<BR>
+	To get to the other side, one man must control the buttons while the other one must hop from platform to platform. Coordination is required - I don't know what would happen if a platform is lowered from beneath your feet, and frankly I'd rather not.</i>"}
+
+/obj/effect/landmark/water_puzzle
+	name = "water puzzle sewers"
+
+/turf/simulated/floor/beach/water/deep/teleport
+	var/turf/teleport_destination
+
+/turf/simulated/floor/beach/water/deep/teleport/Entered(atom/movable/AM)
+	..()
+
+	if(!teleport_destination)
+		var/obj/effect/landmark/water_puzzle/WP = locate(/obj/effect/landmark/water_puzzle) in get_area(src)
+		if(WP)
+			teleport_destination = get_turf(WP)
+		else
+			teleport_destination = src
+
+	if(istype(AM, /obj/item) || istype(AM, /obj/machinery) || istype(AM, /obj/structure) || isliving(AM))
```

Comment:
Good catch thanks

It's here to prevent effects and landmarks from drowning. It caused me a lot of headache yesterday when the puzzle stopped working after a few button presses